### ISSUE TITLE
USH-015 — Missing X-XSS-Protection HTTP Header

### ISSUE THREAD

**Issue by willdoran** [ushahidi/platform#1610]
### Expected behaviour
- Add the HTTP header in the response as: X-XSS-Protection: 1; mode=block
### Actual behaviour
- The X-XSS-Protection HTTP Header is not set by the webserver. This header enables the Cross-Site Scripting (XSS) filter built into most recent web browser.

## Acceptance criteria: 
- [ ] Add documentation for OSS deployers
- [ ] The server response should have a X-XSS-Protection header setup as: X-XSS-Protection: 1; mode=block

Aha! Link: https://ushahiditeam.aha.io/features/PROD-26

**Comment by rjmackay** [comment]
This should be a simple fix.

For client: add this header via nginx config
For cloud ui: add the header via middleware

**Comment by rowasc** [comment]
@willdoran is this in the correct column? Are we blocked?

**Comment by willdoran** [comment]
@rowasc I'm not 100% sure, I don't think it should be in In Development, I'm moving it to ready for spec. I believe the solution for this is to change our nginx config and add some docs for OSS users.

**Comment by rjmackay** [comment]
@rowasc @willdoran I don't think this bears any further speccing. 
Are we happy to approve this as-is?

**Comment by rowasc** [comment]
I think so. Not much to spec here. 
I'd say APPROVED with 2 as is

**Comment by rjmackay** [comment]
I've added this to platform-cloud-ansible. It's only deployed to v3-qa right now. We can deploy to other environments by running
`ansible-playbook -i hosts/ENV cloud-interface.yml platform-client.yml --start-at-task="copy nginx config file"`

This PR adds the header to the example config files
https://github.com/ushahidi/platform-client/pull/1197

**Comment by tuxpiper** [comment]
@rjmackay 

Yes, these are used in the docker images that can be built from the repo:

https://github.com/ushahidi/platform-release/blob/master/run.sh#L134
https://github.com/ushahidi/platform-release/blob/master/run.sh#L163

And they are bundled in the downloadable tarball, for deployers to take as reference while setting up their systems.

**Comment by rjmackay** [comment]
@rowasc @willdoran all our docs reference apache and nginx configuration templates. I've added the headers into those templates. I think thats probably sufficient and better than messing with the documentation. What do you think?

**Comment by willdoran** [comment]
@rjmackay Yes, I think that works. Ideally, anyone deploying it themselves needs to understand how webservers are configured which is an unfortunate fact of life right now.

**Comment by tuxpiper** [comment]
Re-opening to apply on the CloudFront deployments

**Comment by tuxpiper** [comment]
Implementing this properly in the AWS stack would mean having different CDN distributions for index and assets. 

A blocker for that is modifying the client's build routine so that index.html can reference the assets to a different domain. (i.e. "assets.ushahidi.io/app.XXXXX.js" instead of just "app.XXXXX.js"

**Comment by rjmackay** [comment]
@tuxpiper thats probably sensible to do that anyway but I'm not clear *why* it's required for this ticket?

**Comment by rjmackay** [comment]
@tuxpiper this https://github.com/ushahidi/platform-client/pull/1258 gives you a way to pass in a base URL. We'll need to test and verify if all the assets still work as expected though.

**Comment by tuxpiper** [comment]
> @tuxpiper thats probably sensible to do that anyway but I'm not clear why it's required for this ticket?

Yep sorry I didn't document, I found out about this too much towards the end of my day.

The injection of headers in Cloudfront is done with lambda@edge (l@e), which carries a cost per execution. So while things would work by applying the l@e function to the whole distribution, we would be paying extra for unnecessarily invoking the function for purely static assets.

I tried working around that by having different cache behaviors in a single Cloudfront distribution (one for the  "/index.html" path , configured to invoke l@e, another for the rest), but was sabotaged by the fact that any URL that doesn't match an asset file must be internally redirected to /index.html for angular routing .

Then I remembered we had #1322 open, and back then we though it was a good thing to do anyway, to separate purely static assets to their own domain. I was hoping it would be an easy thing to configure and you did it :) Thanks!
